python/cuml/solvers/qn.pyx

```diff
@@ -743,27 +792,33 @@ class QN(Base,
         Predicts the y for X.
 
         """
+        coefs = self.coef_
+        dtype = coefs.dtype
+        _num_classes, n_cols = coefs.shape
+        # binary case
+        if _num_classes == 1:
+            _num_classes = 2
```

Comment:
Would it be possible add a short explanation here?

Reply:
The code required improvements in this section. I added a separate function with comments.